### PR TITLE
Add docs for MQTT Water Heater integration

### PR DESCRIPTION
**Pull request by hookedonunix**
## Proposed change

This accompanies a change to the MQTT integration to add support for Water Heaters. The functionality is very similar to a simplified climate entity, so I've kept it as close as possible for ease of use.

Let me know if there's more information I can add to make these docs more usable.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/93644

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]: Release needs to made specific

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/water_heater.mqtt.markdown
Comment (hookedonunix) [reply source/_integrations/water_heater.mqtt.markdown]: Would that be `2023.6.0` or will I have to wait until the parent is merged to see what release it is?

Comment (jbouwh) [reply source/_integrations/water_heater.mqtt.markdown]: We missed the beta cut yesterday, so that will be 2023.7.0

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]: Can we document the default values too?

Comment (hookedonunix) [reply source/_integrations/water_heater.mqtt.markdown]: Yup, I've done initial as well

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]: This will violate the linters. Please see how this is done fro mqtt climate.

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]: Same here

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]: And here

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [reply source/_integrations/water_heater.mqtt.markdown]: But weird enough not this time. May be because I used the abbreviations. Further it is a bit weird to see a float here, where there is an integer validator.

Comment (hookedonunix) [reply source/_integrations/water_heater.mqtt.markdown]: Whoops, I was looking at the precision definition. That's fixed

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
```

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
  description: Set the initial target temperature. The default value depends on the temperature unit, and will be 43.3°C or 110°F.
```

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
```

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
  description: Minimum set point available. The default value depends on the temperature unit, and will be 43.3°C or 110°F.
```

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
```

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
  description: Maximum set point available. The default value depends on the temperature unit, and will be 60°C or 140°F.
```

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/water_heater.mqtt.markdown
Comment (jbouwh) [inline source/_integrations/water_heater.mqtt.markdown]:
```suggestion
  description: The MQTT topic on which to listen for the current temperature. A `"None"` value received will reset the current temperature. Empty values (`'''`) will be ignored.
```

**Review by jbouwh — APPROVED** [review]
LGTM Thnx,
Can be merged if parent is merged.

**Review by jbouwh — APPROVED** [review]
Can be merged when parent is merged

**Pull request merged**
